### PR TITLE
App-name fixes for master

### PR DESCRIPTION
**Pull request by Kwiboo**
## Description

This PR contains fixes to make Kodi more app-name friendly:
- Installs `<app-name>.desktop` into `usr/share/applications/` instead of `kodi.desktop`
- Uses `${APP_NAME_LC}` in `PathSetup.cmake` instead of `kodi`
- ~~Generates `KodiConfig.cmake` instead of `${APP_NAME}Config.cmake` as all binary addons looks for a `Kodi` package and `KODI_INCLUDE_DIR`~~

This is the same as #11447 but for master.

## Motivation and Context

Building Kodi using a different app-name (OpenPHT in my case) leaves some parts with kodi in the name or path.

## How Has This Been Tested?

Clean builds of Kodi and binary addons with this PR on Windows, macOS and Ubuntu Xenial.

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
@wsnipex @fetzerch @hudokkow ping

File: CMakeLists.txt [hunk 1]
```diff
@@ -27,6 +27,7 @@ include(cmake/scripts/common/AddOptions.cmake)
 include(cmake/scripts/common/ArchSetup.cmake)
 include(cmake/scripts/common/Macros.cmake)
 include(cmake/scripts/common/ProjectMacros.cmake)
+core_find_versions()
```
Comment (hudokkow) [inline CMakeLists.txt]: Any reason to move this?

**Comment by hudokkow** [comment]
I can't help feeling that this is going in the wrong direction.
Apart from the s/kodi/${APP_NAME}/ and innocuous desktop file, the rest is just taking us where we don't want to be: hardcoding file and path names.
In this case, add-ons are the real problem. We should be able to pass ${APP_NAME} and variants to addons build env.

Ideas are welcome.

File: CMakeLists.txt (hunk 1 above)
Comment (Kwiboo) [inline CMakeLists.txt]: ${APP_NAME_LC} is used in PathSetup.cmake with this commit

**Comment by Kwiboo** [comment]
As a third-party developer I see the Kodi-addons config/includes more like an SDK for binary addons and because of that a hardcoded name makes more sense to me and is what all/most binary addons is currently looking for in their cmake-files.

Since I will only be developing on top on the Krypton branch for next few months and there is other ongoing work in the addons system I will drop the `KodiConfig.cmake instead of ${APP_NAME}Config.cmake` commit in this PR.

**Comment by MartijnKaijser** [comment]
ping @wsnipex

**Comment by wsnipex** [comment]
I agree with @hudokkow 
this reverts changes done lately and is going backwards. Passing the APP_NAME to addons would be trivial, but would require a little change in every addon :(
@notspiff, what do you think?

**Comment by notspiff** [comment]
We cannot really pass this to addons in a fully automated fashion due to chicken-egg problems. We can add a parameter in each buildsys to specify the cmake package name to use in place of Kodi. I agree that is probably something that should be done (over time..)

**Comment by Kwiboo** [comment]
@wsnipex @notspiff unsure if you are referring to the dropped commit or the change to `PathSetup.cmake` in this PR.

To solve the chicken-egg problem @AlwinEsch had a good idea in https://github.com/xbmc/xbmc/pull/11378#issuecomment-270756448
"I thought to find maybe a special development kit name who can hold independent from Kodi, so that add-on must always use a `#include <how-is-your-kit-name/...h>`."

The complete set of changes in #11447 is what I currently need to build Krypton and all binary addons using a different app-name (OpenPHT) without having to patch all binary addons looking for another kit-name, other suggestions are welcomed.
This PR have already dropped all changes related to the binary addon kit-name.

**Comment by wsnipex** [comment]
yes, I was referring to the dropped commit. Whats left here is ok.
jenkins build this please

**Comment by hudokkow** [comment]
jenkins build this please

**Pull request merged**